tests/test_io/bad/using_non_output.catala_en

```diff
@@ -14,7 +14,7 @@ scope B:
   assertion a.a = 0
 ```
 ```catala-test-inline
-$ catala Typecheck 
+$ catala Dcalc
```

Comment:
Aaaaaah interesting, now you have to go to `Dcalc` for this sort of errors to appear. For me the `Typechecking` mode of the Catala compiler should run all the checks that would appear if you were to compile it to at least Dcalc and typecheck. So two solutions : either you remap internally the `Typechecking` backend to go to at `Dcalc`, or we add a linting pass at `Scopelang` that checks the invariants needed at later compilation passes.

Reply:
Duplicating the check seems fragile (at the moment its code is tightly bound to the →dcalc transformation)… but that could make sense if we have a whole family of sanity-checks to run at the same time. I already displaced the type cycle detection because it's not reasonable to run typing before that.

I don't see a good reason not to have `catala Typecheck` run the dcalc conversion just for validation: it feels a bit weird, but probably only to the compiler implementers ;)